Address compiler warnings in cuda compilation

At the moment there are several warnings when compiling the cuda stuff. This should be investigated and removed.

Warnings: 

```
Using /tmp/chofer_torch_extensions as PyTorch extensions root...
Creating extension directory /tmp/chofer_torch_extensions/pershom_cuda_ext...
Detected CUDA files, patching ldflags
Emitting ninja build file /tmp/chofer_torch_extensions/pershom_cuda_ext/build.ninja...
Building extension module pershom_cuda_ext...
Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)
[1/6] c++ -MMD -MF vertex_filtration_comp_cuda.o.d -DTORCH_EXTENSION_NAME=pershom_cuda_ext -DTORCH_API_INCLUDE_EXTENSION_H -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/torch/csrc/api/include -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/TH -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/THC -isystem /usr/local/cuda/include -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/include/python3.6m -D_GLIBCXX_USE_CXX11_ABI=0 -fPIC -std=c++14 -c /home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/vertex_filtration_comp_cuda.cpp -o vertex_filtration_comp_cuda.o 
[2/6] c++ -MMD -MF py_bindings.o.d -DTORCH_EXTENSION_NAME=pershom_cuda_ext -DTORCH_API_INCLUDE_EXTENSION_H -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/torch/csrc/api/include -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/TH -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/THC -isystem /usr/local/cuda/include -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/include/python3.6m -D_GLIBCXX_USE_CXX11_ABI=0 -fPIC -std=c++14 -c /home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/py_bindings.cpp -o py_bindings.o 
[3/6] /usr/local/cuda/bin/nvcc -DTORCH_EXTENSION_NAME=pershom_cuda_ext -DTORCH_API_INCLUDE_EXTENSION_H -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/torch/csrc/api/include -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/TH -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/THC -isystem /usr/local/cuda/include -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/include/python3.6m -D_GLIBCXX_USE_CXX11_ABI=0 -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr -gencode=arch=compute_75,code=sm_75 --compiler-options '-fPIC' -std=c++14 -c /home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/tensor_utils.cu -o tensor_utils.cuda.o 
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/tensor_utils.cu: In function ‘void TensorUtils::fill_range_cuda_(at::Tensor)’:
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/tensor_utils.cu:33:27: warning: ‘at::DeprecatedTypeProperties& at::Tensor::type() const’ is deprecated: Tensor.type() is deprecated. Instead use Tensor.options(), which in many cases (e.g. in a constructor) is a drop-in replacement. If you were using data from type(), that is now available from Tensor itself, so instead of tensor.type().scalar_type(), use tensor.scalar_type() instead and instead of tensor.type().backend() use tensor.device(). [-Wdeprecated-declarations]
     auto scalar_type = t.type().scalarType();
                           ^
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:268:1: note: declared here
   DeprecatedTypeProperties & type() const {
 ^ ~~
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/tensor_utils.cu:37:160: warning: ‘T* at::Tensor::data() const [with T = int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
         fill_range_kernel<int32_t><<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(t.data<int32_t>(), t.numel());
                                                                                                                                                                ^
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/tensor_utils.cu:41:162: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
         fill_range_kernel<int64_t><<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(t.data<int64_t>(), t.numel());
                                                                                                                                                                  ^
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
[4/6] /usr/local/cuda/bin/nvcc -DTORCH_EXTENSION_NAME=pershom_cuda_ext -DTORCH_API_INCLUDE_EXTENSION_H -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/torch/csrc/api/include -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/TH -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/THC -isystem /usr/local/cuda/include -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/include/python3.6m -D_GLIBCXX_USE_CXX11_ABI=0 -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr -gencode=arch=compute_75,code=sm_75 --compiler-options '-fPIC' -std=c++14 -c /home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu -o calc_pers_cuda.cuda.o 
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu: In function ‘at::Tensor CalcPersCuda::sorted_pivot_indices_to_merge_pairs_cuda_kernel_call(const at::Tensor&, const at::Tensor&)’:
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu:224:192: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
     extract_slicings_cuda_kernel<<<threads_per_block_apply_slicings, blocks_apply_slicings, 0, at::cuda::getCurrentCUDAStream()>>>(
                                                                                                                                                                                                ^
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu:224:224: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
     extract_slicings_cuda_kernel<<<threads_per_block_apply_slicings, blocks_apply_slicings, 0, at::cuda::getCurrentCUDAStream()>>>(
                                                                                                                                                                                                                                ^
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu:224:266: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
     extract_slicings_cuda_kernel<<<threads_per_block_apply_slicings, blocks_apply_slicings, 0, at::cuda::getCurrentCUDAStream()>>>(
                                                                                                                                                                                                                                                                          ^
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu:245:208: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
     format_extracted_sorted_slicings_to_merge_pairs_kernel<<<threads_per_block, blocks, 0, at::cuda::getCurrentCUDAStream()>>>(
                                                                                                                                                                                                                ^
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu:245:309: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
     format_extracted_sorted_slicings_to_merge_pairs_kernel<<<threads_per_block, blocks, 0, at::cuda::getCurrentCUDAStream()>>>(
                                                                                                                                                                                                                                                                                                                     ^
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu:245:354: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
     format_extracted_sorted_slicings_to_merge_pairs_kernel<<<threads_per_block, blocks, 0, at::cuda::getCurrentCUDAStream()>>>(
                                                                                                                                                                                                                                                                                                                                                                  ^
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu:245:392: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
     format_extracted_sorted_slicings_to_merge_pairs_kernel<<<threads_per_block, blocks, 0, at::cuda::getCurrentCUDAStream()>>>(
                                                                                                                                                                                                                                                                                                                                                                                                        ^
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu: In instantiation of ‘at::Tensor CalcPersCuda::find_slicing_indices_cuda_kernel_call(const at::Tensor&) [with scalar_t = long int]’:
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu:278:79:   required from here
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu:99:157: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
     find_left_slicings_indices_cuda_kernel<scalar_t><<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(
                                                                                                           ~~~~~~~~~~~~~~~                                   ^                                                                                           
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu:99:157: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
     find_left_slicings_indices_cuda_kernel<scalar_t><<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(
                                                                                                           ~~~~~~~~~~~~~~~                                   ^                                                                                           
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu:106:158: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
     find_right_slicings_indices_cuda_kernel<scalar_t><<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(
                                                                                                           ~~~~~~~~~~~~~~~~                                   ^                                                                                           
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu:106:158: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
     find_right_slicings_indices_cuda_kernel<scalar_t><<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(
                                                                                                           ~~~~~~~~~~~~~~~~                                   ^                                                                                           
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu: In instantiation of ‘void CalcPersCuda::merge_columns_cuda_kernel_call(at::Tensor&, const at::Tensor&, int*) [with scalar_t = long int]’:
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu:461:96:   required from here
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu:422:194: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
     merge_columns_cuda_kernel<int64_t><<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(
                                                                                                                                                                                                  ^ 
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu:422:259: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
     merge_columns_cuda_kernel<int64_t><<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(
                                                                                                                                                                                                                                                                   ^ 
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu:422:306: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
     merge_columns_cuda_kernel<int64_t><<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(
                                                                                                                                                                                                                                                                                                                  ^ 
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
[5/6] /usr/local/cuda/bin/nvcc -DTORCH_EXTENSION_NAME=pershom_cuda_ext -DTORCH_API_INCLUDE_EXTENSION_H -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/torch/csrc/api/include -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/TH -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/THC -isystem /usr/local/cuda/include -isystem /scratch2/chofer/opt/anaconda3/envs/pyt_1.6/include/python3.6m -D_GLIBCXX_USE_CXX11_ABI=0 -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr -gencode=arch=compute_75,code=sm_75 --compiler-options '-fPIC' -std=c++14 -c /home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/vr_comp_cuda.cu -o vr_comp_cuda.cuda.o 
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/vr_comp_cuda.cu: In function ‘at::Tensor VRCompCuda::binomial_table(int64_t, int64_t, const c10::Device&)’:
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/vr_comp_cuda.cu:79:157: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
     binomial_table_kernel<<<num_blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(
                                                                                                                                                             ^
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/vr_comp_cuda.cu: In function ‘void VRCompCuda::write_combinations_table_to_tensor(const at::Tensor&, int64_t, int64_t, int64_t, int64_t)’:
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/vr_comp_cuda.cu:248:167: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
     write_combinations_to_tensor_kernel<<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(
                                                                                                                                                                       ^
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/vr_comp_cuda.cu:248:251: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
     write_combinations_to_tensor_kernel<<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(
                                                                                                                                                                                                                                                           ^
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/vr_comp_cuda.cu: In function ‘at::Tensor VRCompCuda::co_faces_from_combinations(const at::Tensor&, const at::Tensor&)’:
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/vr_comp_cuda.cu:341:188: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
     mask_of_valid_co_faces_from_combinations_kernel<<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(
                                                                                                                                                                                            ^
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/vr_comp_cuda.cu:341:261: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
     mask_of_valid_co_faces_from_combinations_kernel<<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(
                                                                                                                                                                                                                                                                     ^
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
/home/pma/chofer/repositories/torchph/torchph/pershom/pershom_cpp_src/vr_comp_cuda.cu:341:304: warning: ‘T* at::Tensor::data() const [with T = long int]’ is deprecated: Tensor.data<T>() is deprecated. Please use Tensor.data_ptr<T>() instead. [-Wdeprecated-declarations]
     mask_of_valid_co_faces_from_combinations_kernel<<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(
                                                                                                                                                                                                                                                                                                                ^
/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/include/ATen/core/TensorBody.h:354:1: note: declared here
   T * data() const {
 ^ ~~
[6/6] c++ py_bindings.o vertex_filtration_comp_cuda.o calc_pers_cuda.cuda.o tensor_utils.cuda.o vr_comp_cuda.cuda.o -shared -L/scratch2/chofer/opt/anaconda3/envs/pyt_1.6/lib/python3.6/site-packages/torch/lib -lc10 -lc10_cuda -ltorch_cpu -ltorch_cuda -ltorch -ltorch_python -L/usr/local/cuda/lib64 -lcudart -o pershom_cuda_ext.so
Loading extension module pershom_cuda_ext...

```

--- a/tests/pershom/test_VRCompCuda.py
+++ b/tests/pershom/test_VRCompCuda.py
@@ -86,7 +86,9 @@ def test_co_faces_from_combinations__no_co_face():
     assert result.tolist() == expected_result
 
 
-@pytest.mark.parametrize("n_vertices, dim_faces", [(10, 2), (8, 3)])
+# Use only rather small parameters for this test. Otherwise, it may cause 
+# random out of memory errors! 
+@pytest.mark.parametrize("n_vertices, dim_faces", [(10, 2), (5, 3)])
 def test_co_faces_from_combinations__result(n_vertices, dim_faces):
     faces = [sorted(f) for f in combinations(range(n_vertices), dim_faces)]
     comb = [sorted(cf) for cf in combinations(range(len(faces)), dim_faces + 1)]

--- a/torchph/pershom/pershom_backend.py
+++ b/torchph/pershom/pershom_backend.py
@@ -90,17 +90,6 @@ except Exception as ex:
     __C = ErrorThrower()
 
 
-def _backend_guard(func):
-    if __C is not None:
-        return func
-
-    else:
-        def raise_error():
-            raise __COMPILATION_ERROR
-
-        return raise_error
-
-
 def find_merge_pairings(
         pivots: Tensor,
         max_pairs: int = -1

--- a/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu
+++ b/torchph/pershom/pershom_cpp_src/calc_pers_cuda.cu
@@ -97,15 +97,15 @@ Tensor find_slicing_indices_cuda_kernel_call(
     const int blocks = pivots.size(0) / threads_per_block + 1;
 
     find_left_slicings_indices_cuda_kernel<scalar_t><<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(
-        pivots.data<scalar_t>(),
-        output.data<scalar_t>(),
+        pivots.data_ptr<scalar_t>(),
+        output.data_ptr<scalar_t>(),
         pivots.size(0));
 
     cudaCheckError(); 
 
     find_right_slicings_indices_cuda_kernel<scalar_t><<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(
-        pivots.data<scalar_t>(),
-        output.data<scalar_t>(),
+        pivots.data_ptr<scalar_t>(),
+        output.data_ptr<scalar_t>(),
         pivots.size(0));
 
     cudaCheckError(); 
@@ -222,9 +222,9 @@ Tensor sorted_pivot_indices_to_merge_pairs_cuda_kernel_call(
     const int threads_per_block_apply_slicings = 256;
     const int blocks_apply_slicings = slicings.size(0) / threads_per_block_apply_slicings + 1;
     extract_slicings_cuda_kernel<<<threads_per_block_apply_slicings, blocks_apply_slicings, 0, at::cuda::getCurrentCUDAStream()>>>(
-        input.data<int64_t>(),
-        slicings.data<int64_t>(),
-        extracted_slicings.data<int64_t>(),
+        input.data_ptr<int64_t>(),
+        slicings.data_ptr<int64_t>(),
+        extracted_slicings.data_ptr<int64_t>(),
         extracted_slicings.size(0),
         extracted_slicings.size(1));
 
@@ -243,12 +243,12 @@ Tensor sorted_pivot_indices_to_merge_pairs_cuda_kernel_call(
     const int blocks = extracted_slicings_sorted.size(0) / threads_per_block + 1;
 
     format_extracted_sorted_slicings_to_merge_pairs_kernel<<<threads_per_block, blocks, 0, at::cuda::getCurrentCUDAStream()>>>(
-        extracted_slicings_sorted.data<int64_t>(),
+        extracted_slicings_sorted.data_ptr<int64_t>(),
         extracted_slicings_sorted.size(0),
         extracted_slicings_sorted.size(1),
-        lengths.data<int64_t>(),
-        row_offset_for_thread.data<int64_t>(),
-        merge_pairings.data<int64_t>());
+        lengths.data_ptr<int64_t>(),
+        row_offset_for_thread.data_ptr<int64_t>(),
+        merge_pairings.data_ptr<int64_t>());
 
     cudaCheckError(); 
 
@@ -420,10 +420,10 @@ void merge_columns_cuda_kernel_call(
     comp_desc_sort_ba.index_fill_(0, targets, -1);
 
     merge_columns_cuda_kernel<int64_t><<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(
-        comp_desc_sort_ba.data<int64_t>(),
+        comp_desc_sort_ba.data_ptr<int64_t>(),
         comp_desc_sort_ba.size(1),
-        cache.data<int64_t>(),
-        merge_pairings.data<int64_t>(),
+        cache.data_ptr<int64_t>(),
+        merge_pairings.data_ptr<int64_t>(),
         merge_pairings.size(0),
         d_boundary_array_needs_resize);
 

--- a/torchph/pershom/pershom_cpp_src/tensor_utils.cu
+++ b/torchph/pershom/pershom_cpp_src/tensor_utils.cu
@@ -30,15 +30,15 @@ void fill_range_cuda_(Tensor t)
     const int threads_per_block = 256;
     const int blocks = t.numel()/threads_per_block + 1;
 
-    auto scalar_type = t.type().scalarType();
+    auto scalar_type = t.scalar_type();
     switch(scalar_type)
     {
         case ScalarType::Int: 
-        fill_range_kernel<int32_t><<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(t.data<int32_t>(), t.numel());
+        fill_range_kernel<int32_t><<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(t.data_ptr<int32_t>(), t.numel());
         break;
 
         case ScalarType::Long: 
-        fill_range_kernel<int64_t><<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(t.data<int64_t>(), t.numel());
+        fill_range_kernel<int64_t><<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(t.data_ptr<int64_t>(), t.numel());
         break;
         
         default:

--- a/torchph/pershom/pershom_cpp_src/vr_comp_cuda.cu
+++ b/torchph/pershom/pershom_cpp_src/vr_comp_cuda.cu
@@ -77,7 +77,7 @@ Tensor binomial_table(int64_t max_n, int64_t max_k, const Device& device){
 
 
     binomial_table_kernel<<<num_blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(
-        ret.data<int64_t>(),
+        ret.data_ptr<int64_t>(),
         max_n, 
         max_k);      
 
@@ -246,13 +246,13 @@ void write_combinations_table_to_tensor(
     int blocks = n_max_over_r/(threads_per_block*n_comb_by_thread) + 1;
 
     write_combinations_to_tensor_kernel<<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(
-        out.data<int64_t>(), 
+        out.data_ptr<int64_t>(), 
         out_row_offset, 
         out.size(1), 
         additive_constant, 
         max_n, 
         r, 
-        bt.data<int64_t>(),
+        bt.data_ptr<int64_t>(),
         n_comb_by_thread, 
         n_max_over_r
     );
@@ -304,9 +304,8 @@ __global__ void mask_of_valid_co_faces_from_combinations_kernel(
             }
         }
 
-
         thrust::device_ptr<int64_t> dptr_tmp(tmp);
-        thrust::sort(thrust::seq, dptr_tmp, dptr_tmp + n_faces+1);
+        thrust::sort(thrust::seq, dptr_tmp, dptr_tmp + n_faces + 1);
 
         bool is_boundary = true; 
         for (int i = 1; i < n_faces+1; i += 2){
@@ -339,12 +338,12 @@ Tensor co_faces_from_combinations(
     int blocks = combinations.size(0)/threads_per_block + 1;
 
     mask_of_valid_co_faces_from_combinations_kernel<<<blocks, threads_per_block, 0, at::cuda::getCurrentCUDAStream()>>>(
-        combinations.data<int64_t>(), 
+        combinations.data_ptr<int64_t>(), 
         combinations.size(0),
         combinations.size(1),
-        faces.data<int64_t>(),
+        faces.data_ptr<int64_t>(),
         faces.size(1), 
-        mask.data<int64_t>()
+        mask.data_ptr<int64_t>()
     );
     
     cudaCheckError(); 

